stdlib/REPL/src/REPL.jl

```diff
@@ -557,9 +584,14 @@ function print_response(errio::IO, response, show_value::Bool, have_color::Bool,
                 if val !== nothing && show_value
                     try
                         if specialdisplay === nothing
-                            Base.invokelatest(display, val)
+                            # display calls may require being run on the main thread
+                            eval_with_backend(backend) do
+                                Base.invokelatest(display, val)
+                            end
                         else
-                            Base.invokelatest(display, specialdisplay, val)
+                            eval_with_backend(backend) do
+                                Base.invokelatest(display, specialdisplay, val)
+                            end
                         end
                     catch
```

Comment:
```suggestion
                    iserr && throw(val)
                    catch
```
(you can probably think of better ways to do this than try catch, but that was the easiest way to write the suggestion to avoid changing indents)